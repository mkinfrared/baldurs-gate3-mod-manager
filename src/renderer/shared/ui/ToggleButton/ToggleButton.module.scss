@import "../../styles/button-base";

.ToggleButton {
  @extend %button-base;

  // TODO ask Natasha about button width
  justify-content: flex-start;

  &:is(.active:not(:disabled)) {
    --border-color: var(--secondary-4);
    --background: var(--secondary-6);

    cursor: auto;
    pointer-events: none;
  }

  .content {
    align-items: center;
    display: inline-flex;
    gap: 1rem;
    max-height: var(--text-height);
    white-space: nowrap;
  }
}
