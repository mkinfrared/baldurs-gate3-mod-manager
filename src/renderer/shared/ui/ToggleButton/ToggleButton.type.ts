import { MouseEventHandler, ReactNode } from "react";

export type ToggleButtonProps = {
  active?: boolean;
  children?: ReactNode;
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  disabled?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
};
