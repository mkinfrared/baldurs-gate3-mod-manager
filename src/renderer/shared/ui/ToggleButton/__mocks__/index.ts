export { default } from "./ToggleButton.mock";
