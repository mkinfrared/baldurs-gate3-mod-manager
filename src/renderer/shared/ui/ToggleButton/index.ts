export { default } from "./ToggleButton";

export * from "./ToggleButton.type";
