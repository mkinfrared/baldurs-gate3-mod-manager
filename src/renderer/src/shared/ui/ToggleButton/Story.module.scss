.Story {
  display: grid;
  gap: 20px;
}
