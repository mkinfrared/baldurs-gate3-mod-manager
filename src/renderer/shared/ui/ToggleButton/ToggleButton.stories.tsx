import { Meta, StoryObj } from "@storybook/react";

import Exhibition from "renderer/shared/assets/icons/exhibition.svg";
import Text from "renderer/shared/ui/Text";

import css from "./Story.module.scss";
import ToggleButton from "./ToggleButton";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/ToggleButton",
  component: ToggleButton,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} satisfies Meta<typeof ToggleButton>;

const Default: Story = {
  args: {
    children: (
      <>
        <Exhibition />
        <Text>Text</Text>
      </>
    ),
  },
  render: (args) => (
    <div className={css.Story}>
      <div>
        <ToggleButton {...args} />
      </div>
      <div>
        <ToggleButton>
          <Exhibition />
          <Text>Text</Text>
        </ToggleButton>
      </div>
    </div>
  ),
};

export { Default };

export default meta;
