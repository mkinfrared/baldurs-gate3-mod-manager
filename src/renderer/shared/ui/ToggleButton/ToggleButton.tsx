// import Button from "renderer/shared/ui/Button";
import { classNames } from "renderer/shared/lib/helpers";

import css from "./ToggleButton.module.scss";
import { ToggleButtonProps } from "./ToggleButton.type";

const ToggleButton = ({
  active,
  className,
  children,
  disabled,
  onClick,
}: ToggleButtonProps) => (
  <button
    className={classNames(
      css.ToggleButton,
      css.contained,
      css.secondary,
      active && css.active,
      className,
    )}
    data-testid="ToggleButton"
    disabled={disabled}
    onClick={onClick}
    type="button"
  >
    <span className={css.content}>{children}</span>
  </button>
);

export default ToggleButton;
