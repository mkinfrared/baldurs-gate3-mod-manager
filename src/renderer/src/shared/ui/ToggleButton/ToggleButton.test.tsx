import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import ToggleButton from "./ToggleButton";

describe("<ToggleButton />", () => {
  const Component = <ToggleButton />;

  it("should be defined", () => {
    expect(ToggleButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("ToggleButton");

    expect(element).toBeDefined();
  });
});
