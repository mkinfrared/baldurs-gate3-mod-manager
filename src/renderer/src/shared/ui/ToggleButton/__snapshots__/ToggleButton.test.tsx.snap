// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ToggleButton /> > should match the snapshot 1`] = `
<div>
  <button
    class="ToggleButton contained secondary"
    data-testid="ToggleButton"
    type="button"
  >
    <span
      class="content"
    />
  </button>
</div>
`;
